integration/embed_test.go

```diff
@@ -115,3 +117,34 @@ func setupEmbedCfg(cfg *embed.Config, curls []url.URL, purls []url.URL) {
 	}
 	cfg.InitialCluster = cfg.InitialCluster[1:]
 }
+
+func TestErrOnEmbedClose(t *testing.T) {
+	defer testutil.AfterTest(t)
+	dir := path.Join(os.TempDir(), fmt.Sprintf("embed-etcd"))
+	defer os.RemoveAll(dir)
+	cfg := embed.NewConfig()
+	cfg.Dir = dir
+	etcd, err := embed.StartEtcd(cfg)
+	if err != nil {
+		t.Fatal(err)
+	}
+	select {
+	case <-etcd.Server.ReadyNotify():
+	case <-time.After(10 * time.Second):
+		etcd.Server.Stop() // trigger a shutdown
+		t.Fatal(fmt.Errorf("Etcd server took too long to start!"))
+	}
+	donec := make(chan struct{})
+	go func() {
```

Comment:
this shouldn't be a separate goroutine since there's no concurrency; the `select` can run directly in the test function's goroutine